crates/sui-network/src/state_sync/mod.rs

```diff
@@ -1187,9 +1187,10 @@ async fn sync_checkpoint_contents_from_archive_iteration<S>(
             upper_limit: Some(end),
             ..Default::default()
         };
-        let Ok((executor, _exit_sender)) = setup_single_workflow(
+        let Ok((executor, _exit_sender)) = setup_single_workflow_with_options(
             StateSyncWorker(store, metrics),
             ingestion_url.clone(),
+            archive_config.remote_store_options.clone(),
```

Comment:
we can add some validation check.
Either check that remote store options are not empty, or explicitly temporarily forbid the use of checkpoints.mainnet.sui.io